pandas/tests/series/methods/test_replace.py

```diff
@@ -305,17 +307,18 @@ def test_replace_mixed_types_with_string(self):
         "categorical, numeric",
         [
             (pd.Categorical(["A"], categories=["A", "B"]), [1]),
-            (pd.Categorical(("A",), categories=["A", "B"]), [1]),
-            (pd.Categorical(("A", "B"), categories=["A", "B"]), [1, 2]),
+            (pd.Categorical(["A", "B"], categories=["A", "B"]), [1, 2]),
         ],
     )
     def test_replace_categorical(self, categorical, numeric):
-        # GH 24971
-        # Do not check if dtypes are equal due to a known issue that
-        # Categorical.replace sometimes coerces to object (GH 23305)
-        s = pd.Series(categorical)
-        result = s.replace({"A": 1, "B": 2})
-        expected = pd.Series(numeric)
+        # GH 24971, GH#23305
+        ser = pd.Series(categorical)
+        result = ser.replace({"A": 1, "B": 2})
+        expected = pd.Series(numeric).astype("category")
```

Comment:
umm why is this a resulting categorical?

Reply:
bc that's what someone implemented for the Categorical.replace behavior.  im on the fence about it, but for now i think we need to be consistent with it

Reply:
hmm ok let's open an issue about this, i dont' think we should raise (or coerce to object) rather than return a new categorical (but maybe others disagree)

Reply:
can you clarify?  based on your earlier comment i expected the opposite opinion, so i think the "dont" may be an unintentional double-negative.